cranelift-wasm/src/translation_utils.rs

```diff
@@ -49,6 +49,11 @@ entity_impl!(MemoryIndex);
 pub struct SignatureIndex(u32);
 entity_impl!(SignatureIndex);
 
+/// Index type of local variable inside the WebAssembly module.
+#[derive(Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Debug)]
+pub struct LocalIndex(u32);
+entity_impl!(LocalIndex);
```

Comment:
This newtype is something I'm slightly hesitant about adding, but it seemed like it was a more consistent choice to have `ModuleEnvironment<'_>::declare_local_name` use a `LocalIndex` newtype rather than a bare `u32` for the local index. Thoughts?

Reply:
**note:**  I'm split the local subsection work out into a separate commit, so if that is overkill I'm happy to revert back to `10f990c` and just handle function names and the module name.

Reply:
I think it's nicer to have a small entity type for this, because this documents the `declare_local_name` method better and makes sure we knowingly pass a `LocalIndex` there.